packages/web3-eth-contract/src/contract.ts

```diff
@@ -130,7 +130,11 @@ type ContractBoundMethod<
 	Abi extends AbiFunctionFragment,
 	Method extends ContractMethod<Abi> = ContractMethod<Abi>,
 > = (
-	...args: Method['Inputs'] extends undefined | unknown ? any[] : Method['Inputs']
+	...args: Abi extends undefined
+		? any[]
+		: Method['Inputs'] extends never
+		? any[]
+		: Method['Inputs']
```

Comment:
This suggestion is exactly the same but avoid nested trinary:
```suggestion
	...args: Method['Inputs'] extends undefined | never ? any[] : Method['Inputs']
```

Reply:
it will not work.. as you see first param is Abi  `Abi extends undefined` and second one Method['Inputs'] `Method['Inputs'] extends never`